engine/gamesys/src/gamesys/components/comp_collection_proxy.cpp

```diff
@@ -130,10 +130,13 @@ namespace dmGameSystem
     {
         CollectionProxyWorld* proxy_world = new CollectionProxyWorld();
         CollectionProxyContext* context = (CollectionProxyContext*)params.m_Context;
-        const uint32_t component_count = context->m_MaxCollectionProxyCount;
+        uint32_t component_count = MIN(params.m_MaxComponentInstances, context->m_MaxCollectionProxyCount);
         proxy_world->m_Components.SetCapacity(component_count);
         proxy_world->m_Components.SetSize(component_count);
-        memset(&proxy_world->m_Components[0], 0, sizeof(CollectionProxyComponent) * component_count);
+        if(component_count > 0)
+        {
+            memset(&proxy_world->m_Components[0], 0, sizeof(CollectionProxyComponent) * component_count);
```

Comment:
Why was this memset necessary?

Reply:
I don't know, to be honest.
We use it for some components like label, sprite, sound, collection proxy and don't use for the other.

Reply:
My bad, I thought it was added.

However, I don't like the "if(count)" part (defensive programming)
This is handled by the count parameter to the function.

Reply:
using proxy_world->m_Components.Begin() I don't need any protective code and it works the same way is in the other components (this particular part is outdated, I mark this conversation as resolved, thx)